jvm/src/main/scala/com/nvidia/rapids/ml/RapidsRelationPlugin.scala

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.rapids.ml
+
+import org.apache.commons.logging.LogFactory
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.connect.planner.SparkConnectPlanner
+import org.apache.spark.sql.connect.plugin.RelationPlugin
+import org.apache.spark.connect.{proto => sparkProto}
+import org.apache.spark.sql.connect.ml.rapids.RapidsConnectUtils
+import org.apache.spark.sql.types.{StringType, StructField, StructType}
+
+import java.util.Optional
+import scala.jdk.CollectionConverters.SeqHasAsJava
+
+class RapidsRelationPlugin extends RelationPlugin {
```

Comment:
Does this need to be registered with spark connect via a spark conf?   Would be helpful to add usage info?